addons/mail/models/mail_thread.py

```diff
@@ -2429,15 +2434,19 @@ def _notify_compute_recipients(self, message, msg_vals):
             exept_partner = [r['id'] for r in recipient_data['partners']]
             if author_id:
                 exept_partner.append(author_id)
-            new_pids = self.env['res.partner'].sudo().search([
-                ('id', 'not in', exept_partner),
-                ('channel_ids', 'in', email_cids),
-                ('email', 'not in', [email_from]),
-            ])
-            for partner in new_pids:
-                # caution: side effect, if user has notif type inbox, will receive en email anyway?
+            sql_query = """ select p.id from res_partner p
+                            left join mail_channel_partner mcp on p.id = mcp.partner_id
+                            left join mail_channel c on c.id = mcp.channel_id
+                            left join res_users u on p.id = u.partner_id
+                                where (u.notification_type != 'inbox' or u.id is null)
```

Comment:
I think it's possible to have different users related to same partner, so technically a partner can have more than one user.
(usually, in practice, could happen that all except one user are archived for a given "user" partner)